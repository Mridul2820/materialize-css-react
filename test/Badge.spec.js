import React from 'react';
import { render } from '@testing-library/react';
import Badge from '../src/Badge';

describe('<Badge />', () => {
  test('renders', () => {
    const { container } = render(<Badge>4</Badge>);
    expect(container).toMatchSnapshot();
  });

  test('should output a badge with the className of `badge new` and', () => {
    const { container } = render(<Badge newIcon>3</Badge>);
    expect(container.querySelector('span.badge.new')).toBeTruthy();
  });

  test('should output a custom caption if provided', () => {
    const { container } = render(<Badge caption="custom caption">4</Badge>);
    expect(
      container.querySelector('span[data-badge-caption="custom caption"]')
    ).toBeTruthy();
  });
});
