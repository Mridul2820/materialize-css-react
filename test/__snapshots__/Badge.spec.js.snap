// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> renders 1`] = `
<div>
  <span
    class="badge"
  >
    4
  </span>
</div>
`;
